### PR TITLE
[feat] Add API functionality to call Seer's test generation endpoint

### PR DESCRIPTION
**Pull request by rohitvinnakota-codecov**
### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

**Review by jennmueng — APPROVED** [review]
lgtm

File: src/sentry/api/endpoints/ai_unit_test_generation.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        )
+
+        response.raise_for_status()
+
+        return response.json().get("run_id")
+
+    def post(
+        self,
+        request: Request,
+        *args,
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/ai_unit_test_generation.py:101]: don't need to use kwargs for the variables you want to access -- can just add them as keywords

File: src/sentry/api/endpoints/ai_unit_test_generation.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        external_id = kwargs.get("external_id")
+        created_at = datetime.now().isoformat()
+
+        if pull_request_number is None:
+            return self._respond_with_error("Missing pull request number.", 400)
+
+        try:
+            pr_id = int(pull_request_number)
+        except ValueError:
+            return self._respond_with_error("Invalid pull request number.", 400)
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/ai_unit_test_generation.py:116]: please use a `Validator`. I would recommend not using `_respond_with_error`. Can see how errors are handled on other endpoints.

**Review by JoshFerge — CHANGES_REQUESTED** [review]
small nits around how APIs are usually done -- should just raise the error, don't need to write your own custom error response function. can poke around a few other sentry endpoints to see how we usually do those things. should also use a validator instead of putting validation logic in the endpoint body.

the primary concern i have is about security -- how do we check authorization and prevent someone passing in repo names / github orgs that their team doesnt own.

are there any endpoints that do similar things in sentry?

File: src/sentry/api/endpoints/ai_unit_test_generation.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.seer.signed_seer_api import get_seer_salted_url, sign_with_seer_secret
+from sentry.types.ratelimit import RateLimit, RateLimitCategory
+from sentry.users.models.user import User
+
+logger = logging.getLogger(__name__)
+
+from rest_framework.request import Request
+
+TIMEOUT_SECONDS = 60 * 30  # 30 minutes
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/ai_unit_test_generation.py:24]: `timedelta(minutes=30)`.

File: src/sentry/api/endpoints/ai_unit_test_generation.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                "Failed to send test generation request to seer",
+                extra={
+                    "created_at": created_at,
+                    "exception": e,
+                },
+            )
+            return self._respond_with_error(
+                "Test generation failed to start.",
+                500,
+            )
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/ai_unit_test_generation.py:137]: you shouldn't need to manually do any of this. errors are already handled and 500s logged at the middleware level https://github.com/getsentry/sentry/blob/07396b7608ee6a7857a35641626b70027f6bc95d/src/sentry/api/base.py#L324

**Review by JoshFerge — COMMENTED** [review]
seems like sentry has linked Repositories as a concept. may want to see how these are used and if checking that a linked repository exists could be a good idea https://github.com/getsentry/sentry/blob/07396b7608ee6a7857a35641626b70027f6bc95d/src/sentry/models/repository.py#L28

**Comment by JoshFerge** [comment]
> Thanks for the review @JoshFerge
> 
> The consumer of this API will be [this](https://github.com/getsentry/getsentry/pull/15270) class
> 
> Regarding security: The class above^ will do validation on the incoming request(which is fired by github). It verifies the signature of the request, as well as whether the user has write access to the repo they are specifying. This is the only route through which this API will be called.
> 
> Regarding custom errors: This API is called via an LLM interface, and the user expects some form of human readable response. The LLM basically takes the response from this API and converts it into a "chat message" format so I prefer to add some context here.
> 
> Regarding Repository object: Seer has a model defined for the Repo object as well and the params passed are validated by that model.
> 
> I switched to using keywords and a validator.
> 
> Please let me know if you have any questions, these changes are time sensitive so I'm always open to creating follow ups to clean things up.

do we have a spec for this anywhere? I'm confused on the flow of this endpoint being called

**Pull request closed without merging**